lib/mergeDeepWith/index.js

```diff
@@ -0,0 +1,10 @@
+import _curry3 from '../_internal/_curry3'
+import _merge from '../_internal/_merge'
+
+export default _curry3(function mergeDeepWith(func, left, right) {
+  var cb = function(l, r) {
+    if (typeof l === 'object' && typeof r === 'object' && !Array.isArray(l)) return mergeDeepWith(func, l, r)
```

Comment:
I would also suggest to reduce the amount of strings as they are not minified
instead use 
```
const tl = typeof l
return tl === 'object' && typeof r === tl && ...
```

Reply:
Make sense

Reply:
Its weight is greater than we supposed to think. Do not do that.